### ISSUE TITLE
⚠️ KerasNLP is becoming KerasHub ⚠️

### ISSUE THREAD

**Issue by mattdangerw** [keras-team/keras-nlp#1831]
## tl;dr 

- We are consolidating KerasNLP and KerasCV into a new **KerasHub** package.
- We will be renaming this GitHub repository to `keras-hub`.
- **All existing usages will continue to work!**
  - You can keep running `pip install keras-nlp`, `keras_nlp` imports will not break. Same with `keras_cv`.

## What's happening

Popular pretrained models are frequently becoming multi-modal. In the text domain, chat LLMs are adding support for image and audio inputs and outputs. In the vision domain, using text encoders is common for a wide range of tasks form image segmentation to image generation.

We do not believe any particular division of pretrained models will help the Keras ecosystem going forward. Distinctions between model architectures (transformers vs convnets vs diffision models) become fuzzy, as do divisions between modalities.

By consolidating to a single library that focuses on easy-to-use, pretrained architectures and weights, we can better deliver a set of features that apply to all pretrained models—easy model publishing and sharing, PEFT and quantization support, scaled up muli-host training.

To that end, we are consolidating KerasNLP and KerasCV into a KerasHub package.

## The plan

We will start start by simply renaming the KerasNLP library to KerasHub. KerasNLP already supports multiple modalities with models like `PaliGemma`, as well as many features (easy model publishing), we would like to bring to CV models.

- This repo will be renamed to KerasHub. The library symbols will be renamed from `keras_nlp` to `keras_hub`.
- We will keep a `keras_nlp` package with all the old imports. This change will be fully backward compatible.
- To move from `keras_nlp` to `keras_hub`, you should be able to simply find and replace all instances of `keras_nlp` with `keras_hub` in code.
- KerasCV models will move to KerasHub on an ongoing bases. You can expect some usage changes to KerasCV models, as we make sure we deliver a consistent UX across modalities.

## When is this happening

We will begin renaming the repository and code in the coming weeks of September, with the first official release of KerasHub likely coming in October. Stay tuned for more details, this issue will be updated with any news.

## Feedback and help

We would appreciate any feedback from the community on this! Please feel free to use this issue to send us thoughts.

We have a number of issues related to the port with the [contributions welcome](https://github.com/keras-team/keras-nlp/issues?q=is%3Aissue+is%3Aopen+label%3A%22stat%3Acontributions+welcome%22) tag.

**Comment by mattdangerw** [comment]
Update: we have now renamed the repository and code to `keras-hub`, but we have not yet released the KerasHub package.

If you would like to try things out as we build, use the nightly package: ` pip install keras-hub-nightly`. If you want the backwards compatible `keras_nlp` imports, you can try that on nightly too: `pip install keras-nlp-nightly`.

Lastly, if you were building the package from source, and want to keep things working with the old imports, you will need to install as follows:

```
!pip install "git+https://github.com/keras-team/keras-hub" "git+https://github.com/keras-team/keras-hub#subdirectory=keras_nlp"
```

That will install both the new `keras_hub` source, as well as a `keras_nlp` shim package.

As always, feedback welcome!

**Comment by mattdangerw** [comment]
Final update (forgot to post it here!), but KerasHub is release as the new stable version. KerasNLP usages should continue to work unaltered. Please let us know if you have any issues.

**Comment by shamiquekhan** [comment]
how's this going? any more help needed? . I would love to contibute .

**Comment by laxmareddyp** [comment]
Hi @divyashreepathihalli ,

Do we still need to keep this issue open?

Thanks.
